proseco/guide.py

```diff
@@ -281,11 +281,35 @@ def run_search_stages(self):
         self.log('Done with search stages')
         stage_cands = cand_guides[cand_guides['stage'] != -1]
         stage_cands.sort(['stage', 'mag'])
+        stage_cands = self.exclude_overlaps(stage_cands)
         guides = self.select_catalog(stage_cands[0:n_guide + GUIDE.surplus_stars])
         if len(guides) < self.n_guide:
             self.log(f'Could not find {self.n_guide} candidates after all search stages')
         return guides
 
+    def exclude_overlaps(self, stage_cands):
+        """
+        Review the stars selected at any stage and exclude stars that overlap in
+        tracking space with another. Overlap is defined as 60 arcsecs in Y and Z.
+        """
+        self.log(f'Checking for guide star overlap in stage-selected stars')
+        nok = np.zeros(len(stage_cands)).astype(bool)
+        for idx, star in enumerate(stage_cands):
+            for jdx, other_star in enumerate(stage_cands):
+                if idx <= jdx:
```

Comment:
This could use a comment.  Since these should be in priority order by mag, I just went with the strategy of excluding the fainter star in a pair.

Reply:
I added a comment.  There are a bunch of ways to make the loop more clear but I didn't care about optimizing.